sdk/testutil/testlog.go

```diff
@@ -15,12 +17,31 @@ func init() {
 	sendTestLogsToStdout = os.Getenv("NOLOGBUFFER") == "1"
 }
 
-func TestLogger(t testing.TB) *log.Logger {
-	return log.New(&testWriter{t}, t.Name()+": ", log.LstdFlags)
+func NewDiscardLogger() hclog.Logger {
+	return hclog.New(&hclog.LoggerOptions{
+		Level:  0,
+		Output: ioutil.Discard,
+	})
+}
+
+func Logger(t testing.TB) hclog.InterceptLogger {
+	return LoggerWithOutput(t, &testWriter{t})
+}
+
+func LoggerWithOutput(t testing.TB, output io.Writer) hclog.InterceptLogger {
+	return hclog.NewInterceptLogger(&hclog.LoggerOptions{
+		Name:   t.Name(),
+		Level:  hclog.Debug,
+		Output: output,
+	})
 }
 
-func TestLoggerWithName(t testing.TB, name string) *log.Logger {
```

Comment:
Similarly for backwards compatibility we should retain the old method and just mark it deprecated:

```
// Deprecated: use LoggerWithName(t)
func TestLoggerWithName(t testing.TB, name string) *log.Logger {
```